packages/flutter_tools/lib/src/build_system/targets.dart

```diff
@@ -0,0 +1,329 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../asset.dart';
+import '../base/file_system.dart';
+import '../base/io.dart';
+import '../base/process_manager.dart';
+import '../build_info.dart';
+import '../compile.dart';
+import '../devfs.dart';
+import '../project.dart';
+
+import 'build_system.dart';
+
+/// Build definitions.
+
+/// List all asset files in a project by parsing the asset manfiest.
+List<File> listAssets(Environment environment) {
+  final AssetBundle assetBundle = AssetBundleFactory.instance.createBundle();
+  assetBundle.build(
+    manifestPath: environment.projectDir.childFile('pubspec.yaml').path,
+    packagesPath: environment.projectDir.childFile('.packages').path,
+  );
+  final List<File> results = <File>[];
+  final Iterable<DevFSFileContent> files = assetBundle.entries.values.whereType<DevFSFileContent>();
+  for (DevFSFileContent devFsContent in files) {
+    results.add(fs.file(devFsContent.file.path));
+  }
+  return results;
+}
+
+/// List all output files in a project by parsing the asset manfiest and
+/// replacing the path with the output directory.
+List<File> listOutputAssets(Environment environment) {
+  final List<File> inputs = listAssets(environment);
+  final List<File> results = <File>[];
+  for (File input in inputs) {
+    final String assetName = fs.path.join(
+      environment.buildDir.path,
+      'flutter_assets',
+      fs.path.basename(input.path),
+    );
+    results.add(fs.file(assetName));
+  }
+  return results;
+}
+
+/// Assemble the assets used in the application into a build directory.
+const Target copyAssets = Target(
+  name: 'copy_assets',
+  inputs: <Object>[
+    '{PROJECT_DIR}/pubspec.yaml',
+    listAssets,
```

Comment:
It would be good to find some way for the API user to provide a pair of functions like (listAssets, listOutputAssets) that do the right thing (like talk about the same set of files) by construction.